var/spack/repos/builtin/packages/libzip/package.py

```diff
@@ -51,9 +54,42 @@ def url_for_version(self, version):
         conditional("cmake", when="@1.4:"), conditional("autotools", when="@:1.3"), default="cmake"
     )
 
+    # Required dependencies
+    with when("build_system=cmake"):
```

Comment:
AFAICS, to support both build_systems, you have to remove this exclusive if for `cmake` and activate them for both (just remove the `if` and the indentation of the variants of course)
```suggestion
```
Then, in the subclass for `autotools`, you have to pass those variants to configure similarily (but different of course) like you pass them in the `cmake` case to `cmake.`